### PR TITLE
C++ bot

### PR DESCRIPTION
**Pull request by DigitumDei**
Pulled last years CPlusPlusBotRunner back into the game-runner
Created c++ bot based on the javascript reference bot
RapidJSON used as JSON support

I have built this in a docker container using GCC 6.3. On WLS ubuntu using GCC 5.4 and in Visual Studio 2017.
I have built and run it on the game runner against the javascript reference bot and it plays to completion. Given that it is logically almost identical to the reference bot, it unsurprisingly won approximately 50% of the time.

### CODE REVIEW

**Comment by Mistborn94** [comment]
Hi @DigitumDei 

Can you please change the logic to the bot to be the same as the other starter bots instead of the reference bot? The rules for the bot logic can be found in the [starter bots readme](https://github.com/EntelectChallenge/2019-Worms/tree/develop/starter-bots#starter-bots-submission-rules)

> The bot itself needs to follow some basic rules:
> 1. Read in the state.json (or map.txt) and parse it into a easy to use structure
>      * Details like the commands, worm health, damage, map size, etc should be read from the state file.  It should **NOT** be hardcoded.
> 1. If one of the opponent's worms is within range fire at it
> 1. Otherwise choose a block in a random direction and do one of the following things
>    1. If the chosen block is air, move to that block
>    1. If the chosen block is dirt, dig out that block
>    1. If the chosen block is deep space, do nothing

**Comment by DigitumDei** [comment]
Simplified the bot so that it shoots at in range opponents and otherwise randomly moves. 

Is keeping the json in rapidjson considered acceptable for point 1?

**Comment by Mistborn94** [comment]
I don't have an issue with using rapid json. :)

File: starter-bots/cplusplus/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# C++ bot for Entelect Challenge 2019
+
+This container includes components required to compile and run a c++ project. The start bot was a c++ version of the supplies javascript reference bot
```
Comment (nellstephanj) [inline starter-bots/cplusplus/README.md]: This should refer to the javascript starter bot, not the reference bot.

Other than that everything seems good 😄 just resolve the current merge conflict and we will merge your pull request in as soon as possible.

Comment (DigitumDei) [reply starter-bots/cplusplus/README.md]: Done. Cleaned up the readme as well since it is definitely not based on the reference bot anymore. :)

**Pull request merged**